spinalcordtoolbox/deepgmseg/deepgmseg.py

```diff
@@ -0,0 +1,215 @@
+# coding: utf-8
+
+import json
+import os
+
+import nipy
+import nibabel as nib
+from nipy.io.nifti_ref import nipy2nifti, nifti2nipy
+import numpy as np
+
+
+from spinalcordtoolbox.resample import nipy_resample
+from . import model
+
+
+class DataResource(object):
+    def __init__(self, dirname):
+        """Initialize the resource with the directory
+        name context.
+
+        :param dirname: the root directory name.
+        """
+        path_script = os.path.dirname(__file__)
+        data_dir = os.path.join(path_script,
+                                "..", "..", "data")
+        data_dir = os.path.abspath(data_dir)
+        self.data_root = os.path.join(data_dir, dirname)
+
+    def get_file_path(self, filename):
+        """Get the absolute file path based on the
+        data root directory.
+
+        :param filename: the filename.
+        """
+        return os.path.join(self.data_root, filename)
+
+
+class CroppedRegion(object):
+    """This class holds cropping information about the volume
+    center crop.
+    """
+
+    def __init__(self, original_shape, starts, crops):
+        """Constructor for the CroppedRegion.
+
+        :param original_shape: the original volume shape.
+        :param starts: crop beginning (x, y).
+        :param crops: the crops (x, y).
+        """
+        self.originalx = original_shape[0]
+        self.originaly = original_shape[1]
+        self.startx = starts[0]
+        self.starty = starts[1]
+        self.cropx = crops[0]
+        self.cropy = crops[1]
```

Comment:
Could keep np arrays instead of x/y members?

Reply:
We can use arrays, but this way is easier to read and maintain since we do a calculation with the coordinates later, so I'll keep the coordinate names explicit.